### ISSUE TITLE
stats handler is stuck with `Could not find event`

### ISSUE THREAD

**Issue by richvdh** [matrix-org/synapse#5329]
Observed on a [sytest log](https://23095-22844864-gh.circle-artifacts.com/0/logs/server-0/homeserver.log), this is being spammed out frequently:

```
2019-06-03 15:42:12,466 - synapse.metrics.background_process_metrics - 203 - ERROR - stats.notify_new_event-7107 - Background process 'stats.notify_new_event' threw an exception
Traceback (most recent call last):
  File "/src/synapse/metrics/background_process_metrics.py", line 201, in run
    yield func(*args, **kwargs)
  File "/venv/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/venv/local/lib/python2.7/site-packages/twisted/python/failure.py", line 512, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/src/synapse/handlers/stats.py", line 74, in process
    yield self._unsafe_process()
  File "/venv/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/venv/local/lib/python2.7/site-packages/twisted/python/failure.py", line 512, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/src/synapse/handlers/stats.py", line 99, in _unsafe_process
    yield self._handle_deltas(deltas)
  File "/venv/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1416, in _inlineCallbacks
    result = result.throwExceptionIntoGenerator(g)
  File "/venv/local/lib/python2.7/site-packages/twisted/python/failure.py", line 512, in throwExceptionIntoGenerator
    return g.throw(self.type, self.value, self.tb)
  File "/src/synapse/handlers/stats.py", line 139, in _handle_deltas
    event_content = (yield self.store.get_event(event_id)).content or {}
  File "/venv/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
    result = g.send(result)
  File "/src/synapse/storage/events_worker.py", line 122, in get_event
    raise NotFoundError("Could not find event %s" % (event_id,))
NotFoundError: 404: Could not find event $62:localhost:42381
```

Presumably related to #5157

**Comment by richvdh** [comment]
assuming this is a recent regression: flagging as a release-blocker

**Comment by hawkowl** [comment]
Fixed in #5352
